src/sidebar/services/auth.js

```diff
@@ -245,19 +251,14 @@ export class AuthService extends TinyEmitter {
 
       if (Date.now() > token.expiresAt) {
         // Token has expired, so we need to fetch a new one.
-        const usingGrantToken =
-          typeof serviceConfig(settings)?.grantToken === 'string';
-
+        const grantToken = getGrantToken();
         tokenInfoPromise = refreshAccessToken(token.refreshToken, {
-          // If we are using automatic login via a grant token, do not persist the
-          // initial access token or refreshed tokens.
-          persist: !usingGrantToken,
+          // Only persist tokens if automatic login is not being used.
+          persist: typeof grantToken === 'undefined',
```

Comment:
Why not simply?

```suggestion
          persist: grantToken === undefined,
```

I understand `undefined` can be redefined (before ES5) but this doesn't affect our application.

Reply:
> I understand undefined can be redefined (before ES5) but this doesn't affect our application.

Good point. I think I did this out of habit. I will simplify it.

Reply:
Elsewhere in the codebase we have similar checks that could be simplified, as well as some where `typeof` is still needed to avoid triggering a `ReferenceError` if a variable does not exist.